index.bs

```diff
@@ -176,6 +179,33 @@ run these steps:
 
 </div>
 
+A <dfn export>file system path</dfn> represents a potential location of a
+[=/file system entry=]. For example, this may be to a virtual path that is
+mapped to real location on disk or in memory, may correspond directly to a
+path on the local file system, or may not correspond to any file on disk at all.
+
+The <dfn local-lt=locating>locate</dfn> algorithm takes a [=/file system path=] and
+returns either a [=/file system entry=] or `null`. The steps of this algorithm
+are [=implementation-defined=].
+
+The <dfn local-lt="getting the path">get the path</dfn> algorithm takes a
+[=/file system entry=] and returns a [=/file system path=].
+If [=locating=] a [=/file system path=] |path| returns a [=/file system entry=] |entry|,
+then [=getting the path=] of |entry| must return |path|.
+
+Each [=/file system path=] has an associated <dfn for="file system path">name</dfn> (a [=string=]),
+which represents the last path component of the file path. For example, if
+[=/file system path=] |path| corresponds to a location represented by a
+virtual path of `data/drafts/example.txt` relative to the root directory of an
+[=origin private file system=], |path|'s [=file system path/name=] should be `example.txt`.
```

Comment:
Yeah, since path isn't exposed directly, do we even need this requirement? It seems you have two choices, either treat path as opaque (and not have this requirement), or define a lot more about the path structure.

Maybe what this is trying to say is that if two entries have the same parent and name they should also have the same path? (i.e. that names are unique?)

Reply:
> Maybe what this is trying to say is that if two entries have the same parent and name they should also have the same path? (i.e. that names are unique?)

I think that's already covered by the symmetry (assuming no intermediate operations are run, as Anne pointed out) between "locate" and "get the path" above

I can update this to explicitly mention that an entry's `name` should match the `name` of the path which locates it. WDYT?

Reply:
> Yeah, since path isn't exposed directly, do we even need this requirement? It seems you have two choices, either treat path as opaque (and not have this requirement), or define a lot more about the path structure.

Yeah it would be nice to keep the path opaque. My only concern is with the `resolve()` method. Consider the following scenario:
```
let root = await navigator.storage.getDirectory();
let dir = await root.getDirectoryHandle('test', {create:true});
let file = await dir.getFileHandle('example.txt', {create:true});

await dir.remove({ recursive: true });

// Currently returns ['test', 'example.txt'] on Chromium and Safari
await root.resolve(file); 
```

Locating `file` will return null, so `resolve()` as we've currently defined it doesn't really work. It seems like that method really should to take a path, as well?

Reply:
Yeah, that makes me think we should define a file system path as a list of strings, similar to how the URL Standard does it. We can still say that the actual physical location is implementation-defined.

Reply:
Implemented this suggestion. LMK what you think!